docs/guides/implementation-guide.mdx

```diff
@@ -169,6 +169,14 @@ function verify(
 );
 ```
 
+:::info
+
+During cross chain communication involving PI chains, we have 2 cases:
+- While initiating a contract communication between a core/sdk chain with PI(permissionlessly deployed) chain, the ISM has to be mentioned on the contract of the core/sdk chain, else the chain's ISM would not be recognized by the relayer.
+- While initiating a contract communication between a PI chain and PI chain, the ISM need not be mentioned in any of the contracts as the mailboxes of these chains would have the address of the ISM module by default
```

Comment:
Nice one here, referring to things as "core chain" or "PI chain" should be the norm to avoid confusion. Would also update this here
e.g.  "else the chain's ISM would not be recognized by the relayer" <-- what is "the chain"


```suggestion
There are two cases to consider when reasoning about cross-chain communication involving [PI (permissionless interoperability) chains](../../deploy-hyperlane.mdx).

1. To communicate between a core chain and a PI chain, the ISM has to be mentioned on the contract of the core/sdk chain, else the chain's ISM would not be recognized by the relayer.
2. To communicate between a PI chain and another PI chain, the ISM does not need to be mentioned in any of the contracts, as the mailboxes of these chains would already have the address of the ISM module by default.
```